### DISCUSSION TITLE
How can I generate the same image using ps.generators.rsa method?

### DISCUSSION THREAD

**Discussion by amisatoayak** [PMEAL/porespy#854] [Q&A] [answered]
Dear Professor

I am trying to generate the same image using 'ps.generators.rsa' through random.seed(), but it doesn't work.

_np.random.seed(0)
im = ps.generators.rsa(im_or_shape=size, r=r[0], clearance=-1, volume_fraction=1 - p[0])_ 

However, the seed works in 'ps.generator.overlapping_spheres'.

_np.random.seed(0)
im = ps.generators.overlapping_spheres(size, r=r[0], porosity=p[0])_

I am confused about this. I wish your answer.
Thank you in advance.

**Comment by jgostick** [comment, marked as answer]
Hi @amisatoayak 
If you upgrade your porespy version you'll see that all generators now accepts the `seed` argument, which we added specifically for the issue you're having.  This seed is handled in a way that numba like (numba is used by rsa/random_spheres to make it faster).
